packages/apollo/src/commands/client/__tests__/generate.test.ts

```diff
@@ -191,6 +191,26 @@ describe("client:codegen", () => {
       expect(fs.readFileSync("API.ts").toString()).toMatchSnapshot();
     });
 
+  test
+    .fs({
+      "schema.json": fullSchemaJsonString,
+      "queryOne.graphql": simpleQuery.toString(),
+      "my.config.js": defaultConfig
+    })
+    .command([
+      "client:codegen",
+      "--config=my.config.js",
+      "--target=typescript",
+      "--fileExtension=.d.ts",
+      "outDirectory"
+    ])
+    .it("writes types for typescript with a custom extension", () => {
+      const [filePath] = fs.readdirSync("./outDirectory");
+      const file = fs.readFileSync(`./outDirectory/${filePath}`).toString();
```

Comment:
Can you add a test before here just double checking the file _extension_ is right? In this test, it'd pass no matter the extension.

```suggestion
	  expect(filePath.includes(".d.ts")).toBeTruthy();
      const file = fs.readFileSync(`./outDirectory/${filePath}`).toString();
```